Lib/fontTools/otlLib/builder.py

```diff
@@ -348,6 +348,76 @@ def build(self):
         # buildLookup_
         return self.buildLookup_(subtables)
 
+    def buildFormat2Subtable(self, ruleset, classdefs, chaining=True):
+        st = self.newSubtable_(chaining=chaining)
+        st.Format = 2
+        st.populateDefaults()
+
+        if chaining:
+            (
+                st.BacktrackClassDef,
+                st.InputClassDef,
+                st.LookAheadClassDef,
+            ) = [c.build() for c in classdefs]
+        else:
+            st.ClassDef = classdefs[1].build()
+
+        inClasses = classdefs[1].classes()
+
+        classSets = []
+        for _ in inClasses:
+            classSet = self.newRuleSet_(format=2, chaining=chaining)
+            classSet.populateDefaults()
+            classSets.append(classSet)
+
+        coverage = set()
+        classRuleAttr = self.ruleAttr_(format=2, chaining=chaining)
+
+        for rule in ruleset.rules:
+            rule_table = self.newRule_(format=2, chaining=chaining)
```

Comment:
i'm a bit distracted by the mixing of camelCase and snake_case for variable names within the same block of code, but whatever.